### ISSUE TITLE
Backend API: Integrate and Test API Locally Prior to Migration to Cloudflare for Song Database Management

### ISSUE THREAD

**Issue by JimNikou** [JimNikou/Loopify#6]
This issue focuses on embedding and testing our backend API in a local development environment before migrating the song database management system to Cloudflare. This is a critical step to ensure that the API functions as expected in the local setup, identifying and resolving potential issues before scaling to a cloud-based infrastructure.

Scope of Work:
1) Embed the API: Integrate the API responsible for handling track data into the backend system, ensuring it interacts correctly with the local database.
2) Local Testing: Conduct thorough tests on the API to validate its ability to retrieve, store, and manage song data effectively. This includes testing various endpoints, error handling, repeating downloads and different response formats.
3) Mock Cloudflare Setup: Try and simulate Cloudflare's caching and database interaction in the local environment to assess any potential performance changes.
4) Prepare for Cloudflare Migration: Based on local tests, identify and resolve any potential compatibility or performance issues that could arise when migrating to Cloudflare.
5) Documentation: Document the process, key observations, and any configurations that need to be adjusted for a smooth transition to Cloudflare.

**Comment by npetalid** [comment]
Cloudflare transition is definitely NOT a requirement! I would not spend any cycles on it.

**Comment by JimNikou** [comment]
> Cloudflare transition is definitely NOT a requirement! I would not spend any cycles on it.

Because of the way we have set-up our track retrieval/serve from the database, the Firestore bucket was only for testing purposes since it has restrictions on how much data can be uploaded without payment. Therefore we have decided to implement a simple server, using an old laptop loaded with Cloudfare, so with that we manage to cut the serve time of a new song in half because there is no upload time, only download and secondly the space is of course much more and free. We resorted to this solution since none of the API's that we looked at have mp3 serve capabilities, so that we can stream the song directly from the android application. This way we can also load, prepare and buffer the songs that are already downloaded or are about to be (only the first song he/she is about to listen to that is not downloaded will take some time to load because of the download to the server first), leaving the user with a more responsive experience.
